superset/commands/base.py

```diff
@@ -0,0 +1,44 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from abc import ABC, abstractmethod
+from typing import List, Tuple
+
+from marshmallow.validate import ValidationError
+
+CommandValidateReturn = Tuple[bool, List[ValidationError]]
+
+
+class BaseCommand(ABC):
+    """
+        Base class for all Command like Superset Logic objects
+    """
+
+    @abstractmethod
+    def run(self):
+        """
+        Run executes the command
+        :return:
+        """
+        pass
+
+    @abstractmethod
+    def validate(self) -> CommandValidateReturn:
```

Comment:
I like the idea of making `validate` an explicit external interface. I can see this functionality being useful down the line.

Reply:
That's the idea, not totally happy with the name, since the validates and fetches/populates data. Maybe `fetch_validate`, thoughts?

Reply:
I'd prefer to keep it a single word. It can be understood that `validate` may run queries if necessary. I would prefer not to be too prescriptive on the internal workings of a command. They should be generic interfaces that allow a great deal of flexibility.